src/smtp_server.c

```diff
@@ -157,6 +158,10 @@ int smtp_server_process(struct smtp_server_context *ctx, const char *cmd, const
 		}
 
 		if (ctx->code) {
+			if (cmd != NULL) {
+				js_call("smtpServer", format_func_name(cmd), JSVAL_NULL);
```

Comment:
The string allocated by `format_func_name()` should be freed after calling `js_call()`. Better yet, since this is the only place where `format_func_name()` is used, rename it to 'call_js_handler' (or something similar) and have it call `js_call()` on its own. This way, the JS function name can be stored in a local variable and doesn't need to be freed any more.